### PR TITLE
[14.0][WIP] price per qty

### PR DESCRIPTION
**Pull request by sebastienbeau**
- [x] Add export of price per qty
- [x] fix odoo rounding issue :'(

### CODE REVIEW

**Comment by sebastienbeau** [comment]
@ivantodorovich @simahawk @PierrickBrun  ready for review

**Review by ivantodorovich — APPROVED** [review]
LG 👍

File: shopinvader/models/shopinvader_variant.py [hunk 1]
```diff
@@ -218,10 +218,17 @@ def _get_price(self, qty=1.0, pricelist=None, fposition=None, company=None):
         price_unit = AccountTax._fix_tax_included_price_company(
             price_unit, product.taxes_id, taxes, company
         )
+        # Apply the right precision on price unit
+        price_dp = self.env["decimal.precision"].precision_get("Product Price")
+        # be carefull odoo rounding implementation do not return the shortest
+        # representation of a float this mean if the price_unit is 211.70
+        # See issue: https://github.com/shopinvader/odoo-shopinvader/issues/1041
+        # Odony exemple: https://gist.github.com/odony/5269a695545902e7e23e761e20a9ec8c
+        price_unit = float(float_repr(price_unit, price_dp))
```
Comment (simahawk) [inline shopinvader/models/shopinvader_variant.py]: can you move this to `shopinvader.utils`?

File: shopinvader_price_per_qty/README.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+==============================
+ShopInvader Price per Quantity
+==============================
+
+Thsi module extract the price per quantity in order to make in available in your shopinvader website
```
Comment (simahawk) [inline shopinvader_price_per_qty/README.rst]: can you elaborate a bit on this? It should mention pricelist etc.

Also, please use readme fragments.

File: shopinvader_price_per_qty/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Copyright 2016 Akretion (http://www.akretion.com)
```
Comment (simahawk) [inline shopinvader_price_per_qty/__manifest__.py:1]: I'd move this module to a new PR to fast track the fix for the core

**Review by simahawk — CHANGES_REQUESTED** [review]

File: shopinvader_price_per_qty/models/shopinvader_variant.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                ("product_tmpl_id", "=", self.product_tmpl_id.id),
+                ("categ_id", "=", self.categ_id.id),
+                "&",
+                "&",
+                ("product_id", "=", False),
+                ("categ_id", "=", False),
+                ("product_tmpl_id", "=", False),
+            ]
+        )
+        return sorted({item.min_quantity for item in items if item.min_quantity > 1})
```
Comment (lmignon) [inline shopinvader_price_per_qty/models/shopinvader_variant.py:33]: @sebastienbeau you should take into account the date_start and date_end fields into your domain.

**Review by lmignon — CHANGES_REQUESTED** [review]
Nice functionality but at which price (perf cost?)

**Comment by sebastienbeau** [comment]
@simahawk @lmignon ready !

**Review by ivantodorovich — APPROVED** [review]

File: shopinvader_price_per_qty/tests/test_product.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        if expected is None:
+            self.assertNotIn(10, price["per_qty"])
+        else:
+            self.assertIn(10, price["per_qty"])
+            self.assertEqual(expected, price["per_qty"][10])
+
+    def test_price_with_date(self):
+        yesterday = datetime.now() - timedelta(1)
+        tomorrow = datetime.now() + timedelta(1)
+        self._test_price_for_case(None, None, 456.52)
```
Comment (simahawk) [inline shopinvader_price_per_qty/tests/test_product.py:43]: wouldn't be better to test w/ different expected values?

**Review by simahawk — APPROVED** [review]

**Comment by simahawk** [comment]
@sebastienbeau would be nice to cleanup commits (eg: just 2 commits, 1 for the fix and 1 for the new module)

File: shopinvader_price_per_qty/tests/test_product.py (hunk 5 above)
Comment (sebastienbeau) [reply shopinvader_price_per_qty/tests/test_product.py:43]: I think it's enough, the aim is just to check the behaviour on date (returning different price base on the qty is already tested before)

**Comment by sebastienbeau** [comment]
> Nice functionality but at which price (perf cost?)

@lmignon I think we can merge it, for sure it add extra cost during the recomputation of the product, but odoo is now faster so in our case it work well (we have 78608 products with 10 qty step per product, computation is not fast but totally acceptable)

**Comment by sebastienbeau** [comment]
@lmignon let's merge it ?

**Comment by sebastienbeau** [comment]
@lmignon can you remove you blocking review ?

**Review by lmignon — APPROVED** [review]

**Pull request merged**